src/app/drive/components/DriveExplorer/DriveExplorer.tsx

```diff
@@ -458,31 +466,53 @@ declare module 'react' {
 
 const uploadItems = async (props: DriveExplorerProps, rootList: IRoot[], files: File[]) => {
   const { dispatch, currentFolderId, onDragAndDropEnd } = props;
-  if (files.length) {
-    // files where dragged directly
-    await dispatch(
-      storageThunks.uploadItemsThunkNoCheck({
-        files,
-        parentFolderId: currentFolderId,
-        options: {
-          onSuccess: onDragAndDropEnd,
-        },
-      }),
-    );
-  }
 
-  if (rootList.length) {
-    for (const root of rootList) {
+  let totalFilesToUpload: number;
+
+  totalFilesToUpload = files.length;
+
+  const countTotalItemsToUpload = (rootList) => {
+    rootList.forEach((n) => {
+      totalFilesToUpload += n.childrenFiles.length;
+      if (n.childrenFolders.length >= 1) {
+        countTotalItemsToUpload(n.childrenFolders);
+      }
+    });
+  };
+
+  countTotalItemsToUpload(rootList);
+
+  if (totalFilesToUpload < 1000) {
+    if (files.length) {
+      // files where dragged directly
       await dispatch(
-        storageThunks.uploadFolderThunkNoCheck({
-          root,
-          currentFolderId,
+        storageThunks.uploadItemsThunkNoCheck({
+          files,
+          parentFolderId: currentFolderId,
           options: {
             onSuccess: onDragAndDropEnd,
           },
         }),
       );
     }
+    if (rootList.length) {
+      for (const root of rootList) {
+        await dispatch(
+          storageThunks.uploadFolderThunkNoCheck({
```

Comment:
Same here, why is this change made and how is it related to 1000 items upload limit 🤔 

If it's not related, its ok, just add it to the description

Reply:
I have not replaced anything, I have only wrapped the block that already existed in an `if`